### PR TITLE
Update dependencies and add optics example

### PR DESCRIPTION
**Pull request by chiroptical**
In this PR, I update all of the dependencies and add optics as well as functioning KSP setup.

This update does generate a lot of deprecation warnings, but I could update those in a subsequent PR.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -21,7 +28,7 @@ kotlin {
     nodejs()
   }
 
-  linuxX64()
+  // linuxX64()
```
Comment (chiroptical) [inline build.gradle.kts:31]: This took a really long time on my machine. Is that expected? I commented it out to see if everything else would work (and it did).

File: src/jvmMain/kotlin/UseIngredient.kt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+val lettuce = Ingredient("lettuce")
+val arugula = Ingredient.name.modify(lettuce) { "arugula" }
```
Comment (chiroptical) [inline src/jvmMain/kotlin/UseIngredient.kt:2]: Without `kotlin.srcDir` above, it won't recognize the companion `name` and the compiler will complain.

File: build.gradle.kts [hunk 3]
```diff
@@ -43,22 +50,26 @@ kotlin {
   sourceSets {
     commonMain {
       dependencies {
+        kotlin.srcDir("build/generated/ksp/metadata/commonMain/kotlin")
```
Comment (raulraja) [inline build.gradle.kts:53]: Thanks @chiroptical, Do you need this if you add a package in the Sample file?

Comment (chiroptical) [reply build.gradle.kts:53]: Sorry, what do you mean by "Sample file"? I am pretty sure that is how gradle/IDE finds all of the KSP generated companion objects. TBH, I am not super familiar with gradle and there may be a better way to do this. I just don't have enough Kotlin experience to help there.

This is working on my machine. I can use the optics within and outside `commonMain` with this.

File: build.gradle.kts (hunk 1 above)
Comment (chiroptical) [reply build.gradle.kts]: (this worked after stepping away from this PR for awhile)

**Comment by nomisRev** [comment]
Hey @chiroptical,

Sorry for the late replies. I noticed that it's failing for me locally, but it works if we bump to `1.8.20`.

> kotlin("multiplatform") version "1.8.20" apply true
  id("io.kotest.multiplatform") version "5.5.4" apply true
  id("com.google.devtools.ksp") version "1.8.20-1.0.11"

Do you mind bumping these versions in this PR?

**Pull request merged**